source/models/customizing-serializers.md

```diff
@@ -0,0 +1,430 @@
+have a model with a `hasMany` relationship:
+
+```app/models/post.js
+export default DS.Model.extend({
+  comments: DS.hasMany('comment', { async: true })
+});
+```
+
+The JSON should encode the relationship as an array of IDs:
+
+```js
+{
+  "post": {
+    "comments": [1, 2, 3]
+  }
+}
+```
+
+`Comments` for a `post` can be loaded by `post.get('comments')`. The
+REST adapter will send 3 `GET` request to `/comments/1/`,
+`/comments/2/` and `/comments/3/`.
+
+Any `belongsTo` relationships in the JSON representation should be the
+camelized version of the Ember Data model's name. For example, if you have
+a model:
+
+```app/models/comment.js
+export default DS.Model.extend({
+  post: DS.belongsTo('post')
+});
+```
+
+The JSON should encode the relationship as an ID to another record:
+
+```js
+{
+  "comment": {
+    "post": 1
+  }
+}
+```
+
+If needed these naming conventions can be overwritten by implementing
+the `keyForRelationship` method.
+
+```app/serializers/application.js
+export default DS.RESTSerializer.extend({
+  keyForRelationship: function(key, relationship) {
+    return key + 'Ids';
+  }
+});
+```
+
+#### Sideloaded Relationships
+
+To reduce the number of HTTP requests necessary, you can sideload
+additional records in your JSON response. Sideloaded records live
+outside the JSON root, and are represented as an array of hashes:
+
+```json
+{
+  "post": {
+    "id": 1,
+    "title": "Node is not omakase",
+    "comments": [1, 2, 3]
+  },
+
+  "comments": [{
+    "id": 1,
+    "body": "But is it _lightweight_ omakase?"
+  },
+  {
+    "id": 2,
+    "body": "I for one welcome our new omakase overlords"
+  },
+  {
+    "id": 3,
+    "body": "Put me on the fast track to a delicious dinner"
+  }]
+}
+```
+
+#### Creating Custom Transformations
+
+In some circumstances, the built in attribute types of `string`,
+`number`, `boolean`, and `date` may be inadequate. For example, a
+server may return a non-standard date format.
+
+Ember Data can have new JSON transforms
+registered for use as attributes:
+
+```app/transforms/coordinate-point.js
+export default DS.Transform.extend({
+  serialize: function(value) {
+    return [value.get('x'), value.get('y')];
+  },
+  deserialize: function(value) {
+    return Ember.create({ x: value[0], y: value[1] });
+  }
+});
+```
+
+```app/models/cursor.js
+export default DS.Model.extend({
+  position: DS.attr('coordinatePoint')
+});
+```
+
+When `coordinatePoint` is received from the API, it is
+expected to be an array:
+
+```js
+{
+  cursor: {
+    position: [4,9]
+  }
+}
+```
+
+But once loaded on a model instance, it will behave as an object:
+
+```js
+var cursor = App.Cursor.find(1);
+cursor.get('position.x'); //=> 4
+cursor.get('position.y'); //=> 9
+```
+
+If `position` is modified and saved, it will pass through the
+`serialize` function in the transform and again be presented as
+an array in JSON.
+
+
+
+### JSONSerializer
+
+Not all APIs follow the conventions that the `RESTSerializer` uses
```

Comment:
Maybe these couple paragraphs could be simplified to say that the JSONSerializer is just the same as REST but with no sideloading.